### PR TITLE
create-repo-implementation

### PR DESCRIPTION
**Pull request by ValentynSharshon**

### CODE REVIEW

File: src/main/java/com/gmail/woosay333/onlinebookstore/entity/Book.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Entity
+@Table(name = "books")
+@Data
```
Comment (karma-o) [inline src/main/java/com/gmail/woosay333/onlinebookstore/entity/Book.java]: `@Data` is not recommended to use with the `@Entity` because of probable poor query performance

File: src/main/java/com/gmail/woosay333/onlinebookstore/repository/impl/BookRepositoryImpl.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
```
Comment (karma-o) [inline src/main/java/com/gmail/woosay333/onlinebookstore/repository/impl/BookRepositoryImpl.java]: this is redundant, when you have a single constructor
```suggestion
    private final SessionFactory sessionFactory;

```

File: src/main/resources/db/changelog/changeset/01-create-table-books.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
+databaseChangeLog:
```
Comment (karma-o) [inline src/main/resources/db/changelog/changeset/01-create-table-books.yaml]: I do not think we have liquibase addition as part of the tasks, and also you have it disabled in the `app.properties`, so I'd remove that and other stuff that is related to that for now because it is not used

**Review by karma-o — CHANGES_REQUESTED** [review]
just some minor comments

File: src/main/resources/db/changelog/changeset/01-create-table-books.yaml (hunk 3 above)
Comment (ValentynSharshon) [reply src/main/resources/db/changelog/changeset/01-create-table-books.yaml]: Since the task specifies that you need to use the create-drop parameter, I have disabled the use of Liquibase for now. But I would like to keep it for the future and not delete it.

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+databaseChangeLog:
```
Comment (yevheniisynytsia) [inline src/main/resources/db/changelog/db.changelog-master.yaml]: it is a bad practice to add code that is not related to the task.
For example, you will merge it, then your teammate will use it in another branch. But at the same moment, you decided to change it completely, but another engineer is already dependent on the first version and has to change his code because it was merged early.

**Review by yevheniisynytsia — CHANGES_REQUESTED** [review]

File: src/main/resources/db/changelog/db.changelog-master.yaml (hunk 4 above)
Comment (ValentynSharshon) [reply src/main/resources/db/changelog/db.changelog-master.yaml]: Done 👍
